test/core/event_engine/windows/iocp_test.cc

```diff
@@ -0,0 +1,379 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <grpc/support/port_platform.h>
+
+#include "src/core/lib/event_engine/windows/iocp.h"
+
+#include <thread>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "absl/status/status.h"
+#include "absl/time/time.h"
+#include "absl/types/variant.h"
+
+#include <grpc/grpc.h>
+#include <grpc/support/log_windows.h>
+
+#include "src/core/lib/event_engine/poller.h"
+#include "src/core/lib/event_engine/windows/socket.h"
+#include "src/core/lib/event_engine/windows/windows_engine.h"
+#include "src/core/lib/iomgr/error.h"
+#include "test/core/event_engine/windows/basic_closure.h"
+#include "test/core/event_engine/windows/create_sockpair.h"
+#include "test/core/event_engine/windows/self_deleting_closure.h"
+
+// DO NOT SUBMIT(hork): get sanitizers working
+
+namespace {
+using ::grpc_event_engine::experimental::BasicClosure;
+using ::grpc_event_engine::experimental::CreateSockpair;
+using ::grpc_event_engine::experimental::EventEngine;
+using ::grpc_event_engine::experimental::IOCP;
+using ::grpc_event_engine::experimental::Poller;
+using ::grpc_event_engine::experimental::SelfDeletingClosure;
+using ::grpc_event_engine::experimental::WindowsEventEngine;
+using ::grpc_event_engine::experimental::WinSocket;
+}  // namespace
+
+class IOCPTest : public testing::Test {};
+
+TEST_F(IOCPTest, ClientReceivesNotificationOfServerSend) {
+  auto engine = absl::make_unique<WindowsEventEngine>();
```

Comment:
I'd like to see this test not reference `WindowsEventEngine` - as it stands if there's a bug I must debug the entirety of the `WindowsEventEngine` system, rather than just the slice that I might suspect to be faulty.

Reply:
Done, this now relies on a ThreadedExecutor (identical to the one used inside the WindowsEventEngine, but it does not have to be)